### PR TITLE
Marriage abroad changes Dominican Republic

### PR DESCRIPTION
**Pull request by ikennaokpala**
Pivotal Tracker: https://www.pivotaltracker.com/projects/1270592/stories/109437774

* Replace contact details with appointment booking link.

## Factcheck

https://smart-answers-pr-109437774.herokuapp.com/marriage-abroad/y/dominican-republic/ceremony_country/partner_local/same_sex
https://smart-answers-pr-109437774.herokuapp.com/marriage-abroad/y/dominican-republic/uk/partner_local/same_sex
https://smart-answers-pr-109437774.herokuapp.com/marriage-abroad/y/dominican-republic/third_country/partner_local/same_sex

## Expected Changes
[URL](https://www.gov.uk/marriage-abroad/y/dominican-republic/ceremony_country/partner_british/same_sex)
     * Changed Contact details to appointment link.

### Before

screen shot 2016-02-19 at 16 23 47

### After

screen shot 2016-02-19 at 16 21 12

### CODE REVIEW

**Comment by floehopper** [comment]
Apart from my all minor comments, this looks good to me. :+1:

**Comment by ikennaokpala** [comment]
@floehopper Changes made.. waiting on the factcheck to come through..

**Comment by floehopper** [comment]
:+1:

**Comment by erikse** [comment]
Given we've added Dominican Republic to the regression tests in this [commit](https://github.com/alphagov/smart-answers/commit/aa593ed57ce76b14dde7c5fbae5f8e93ec97086c) it would be great to add a set of expected responses and outcomes to this [file](https://github.com/alphagov/smart-answers/blob/master/test/data/marriage-abroad-responses-and-expected-results.yml).

**Comment by ikennaokpala** [comment]
@erikse Done

**Comment by floehopper** [comment]
Is there anything stopping this PR from being merged?

**Comment by ikennaokpala** [comment]
@floehopper it has changed seem your last review.. do take a look

**Comment by floehopper** [comment]
I've removed the LGTM label from this PR, because @ikennaokpala says it has changed significantly since it was originally reviewed.

File: test/data/marriage-abroad-questions-and-responses.yml [hunk 1]
```diff
@@ -29,6 +29,7 @@
 - cyprus
 - czech-republic
 - denmark
+- dominican-republic
```
Comment (floehopper) [inline test/data/marriage-abroad-questions-and-responses.yml]: While it's good to have small commits, that idea can be taken too far. I prefer to see an atomic & coherent set of changes per commit. I think this commit is probably too small and should probably be combined with others to make a coherent set of changes.

I think it would also help if the commit note explained **why** you are making this change.

Comment (floehopper) [inline test/data/marriage-abroad-questions-and-responses.yml]: Minor: The commit note subject line is too generic.

**Comment by floehopper** [comment]
I found this PR quite hard to follow for the following reasons:

* Some of the earlier commits are too small so that a coherent set of changes is spread over multiple commits which is hard to follow.
* I always try to make all the tests pass after each commit unless that's particularly awkward. I'm not convinced that that would be true of all the commits in this branch.
* Quite a few of the later commits look like fixup commits, but are not marked as such. Do you plan to turn them into fixup commits?
* While I understand the idea of introducing a country-specific outcome, I think the commit where it is introduced is a big jump - I don't really understand where all the code in that template has come from.

In writing some of the above I realise our approach of curating the commits in a PR to look as if we'd done the right thing the first time isn't really documented anywhere. Likewise the idea of trying to have the tests all passing after each commit isn't really explained. I plan to update [CONTRIBUTING.md](https://github.com/alphagov/smart-answers/blob/master/CONTRIBUTING.md) and/or the [Git style guide](https://github.com/alphagov/styleguides/blob/master/git.md) to address this.

I wonder if the following sequence of commits might be easier to understand:

1. Add `dominican-republic` to `marriage-abroad-questions-and-responses.yml`, generate the corresponding additions to `marriage-abroad-responses-and-expected-results.yml`, generate the corresponding additions to the regression test artefacts. All in a single commit.
2. Introduce a country-specific outcome for `dominican-republic` without changing the behaviour i.e. ensure that the regression tests still pass with no changes to the artefacts. Update the regression test checksums accordingly. All in a single commit.
3. A series of commits making any required changes to the behaviour for the Dominican Republic. If these are small enough, you could update the regression test artefacts & checksums within each commit.
4. If necessary, have a single commit at the end which updates the regression test artefacts.
5. If necessary, have a single commit at the end which updates the regression test checksums.

Since @chrisroos is refactoring `marriage-abroad`, I think it would be useful to him to comment on this - if only to see whether the country-specific outcome fits with the direction he's heading in.

Lastly, I think it's really confusing to keep rebasing and force-pushing to a single PR. I think it's better to either (a) non-force-push fixup or squash commits to the existing PR with a final rebase and force-push after review and just before merging; or (b) create a new branch based on the original one and open a new PR to supersede the original one. The advantage of both of these approaches is that the reviewer's comments remain easily accessible. I'll try to write this up somewhere so we're all on the same page.

Sorry I've written loads here. It's just that a few things reminded me of stuff I've been meaning to explain for a while.

**Comment by ikennaokpala** [comment]
@floehopper 

* This PR is my longest running one, thus far, I believe its my first. 
* Yes a majority of commits I plan to fixup..
* I will be creating a new PR that supersedes this.

**Comment by floehopper** [comment]
Ah. I hadn't realised it had been around so long. New PR sounds like a good plan.

**Comment by ikennaokpala** [comment]
This PR is now superseded by #2406

**Comment by chrisroos** [comment]
> Since @chrisroos is refactoring marriage-abroad, I think it would be useful to him to comment on this - if only to see whether the country-specific outcome fits with the direction he's heading in.

@floehopper / @ikennaokpala. Country specific outcomes sound good to me. I've introduced some more of these in a local branch that I'll create a PR for soon.

**Comment by ikennaokpala** [comment]
@chrisroos great.. looking forward to that.. BTW Compliments of the season guys
